katello/hooks/pre/30-upgrade.rb

```diff
@@ -93,5 +115,9 @@ def fail_and_exit(message)
     upgrade_step :migrate_foreman, :run_always => true
   end
 
+  if param_value('foreman', 'db_manage') || param_value('katello', 'candlepin_manage_db')
```

Comment:
Probably good to add a variable `has_local_postgresql` and reuse that.

Reply:
To make it really EL-8 ready you can add `facts[:os][:release][:major] == '7'` as a condition.